dev/reference/garbage-collection/overview.md

```diff
@@ -0,0 +1,45 @@
+---
+title: GC Overview
+summary: Overview of Garbage Collection in TiDB.
+category: reference
+---
+
+# GC Overview
+
+TiDB uses MVCC to control transaction concurrency. When you update the data, the original data is not deleted immediately but is kept together with the new data, with a timestamp to distinguish the version. The goal of Garbage Collection (GC) is to clear the obsolete data.
+
+## GC process
+
+Each TiDB cluster contains a TiDB instance that is selected as the GC leader, which controls the GC process.
+
+GC runs periodically on TiDB. The default frequency is once every 10 minutes. For each GC, TiDB firstly calculates a timestamp called "safe point" (defaults to the current time minus 10 minutes). Then, TiDB clears the obsolete data under the premise that all the snapshots after the safe point retain the integrity of the data. Specifically, there are three steps involved in the GC process:
+
+1. Resolve Locks
+2. Delete Ranges
+3. Do GC
+
+### Resolve Locks
+
+The TiDB transaction model is implemented based on [Google's Percolator](https://ai.google/research/pubs/pub36726). It's mainly a two-phase commit protocol with some practical optimizations. When the first phase is finished, all the related keys are locked. Among these locks, one is the primary lock and the others are secondary locks which contain a pointer to the primary lock; in the secondary phase, the key with the primary lock gets a write record and its lock is removed. The write record indicates the write or delete operation in the history or the transactional rollback record of this key. The type of write record that replaces the primary lock indicates whether the corresponding transaction is committed successfully. Then all the secondary locks are replaced successively. If the threads fail to replace the secondary locks, these locks are retained.
```

Comment:
> " with some practical optimizations" didn't shows in the Chinese doc. Should we also add it to the Chinese doc?

This is the original text that I reused. Since it does not say something specific I feel it's OK to not mention in Chinese.